### PR TITLE
Use explicitly sized unsigned integral data types and add more tests

### PR DESCRIPTION
**Pull request by scheibel**

### CODE REVIEW

File: source/tests/threadingzeug-test/parallel_for_test.cpp [hunk 1]
```diff
@@ -7,31 +7,186 @@
...
+        ASSERT_EQ(vec1[i], vec2[i]);
+    }
+}
+
+TEST_F(parallelFor_test, SizeTCompilation)
+{
+    size_t start = 0;
+    size_t end = 1;
+
+    parallelFor(start, end, [] (int i) {
```
Comment (mjendruk) [inline source/tests/threadingzeug-test/parallel_for_test.cpp]: This line causes an error on Mac. “Call to 'parallelFor' is ambiguous.”

**Comment by scheibel** [comment]
Maybe, the parameter ```i``` has also be of type ```size_t```?

File: source/tests/threadingzeug-test/parallel_for_test.cpp (hunk 1 above)
Comment (scheibel) [inline source/tests/threadingzeug-test/parallel_for_test.cpp]: Maybe, the parameter ```i``` has also be of type ```size_t```?

Comment (sbusch42) [inline source/tests/threadingzeug-test/parallel_for_test.cpp]: http://stackoverflow.com/questions/11603818/why-is-there-ambiguity-between-uint32-t-and-uint64-t-when-using-size-t-on-mac-os

**Pull request merged**